### ISSUE TITLE
How to install Telemeter

### ISSUE THREAD

**Issue by BartEngelen** [thibmaek/custom-components#12]
Replace the text between backticks with your config and tick off the relevant boxes (after creating the issue):

## System details

- Home Assistant version: `0.90.1`
- Host: `Raspberry Pi 3 Model B+`
- Home Assistant setup:
  - [ ] Hassbian / Raspbian
  - [ ] Docker
  - [X] Hass.io
    - [x] HassOS
    - [ ] Other
  - [ ] Python

## Hass.io

If you are using Hass.io, paste the output of following commands in the relevant code blocks. Otherwise you can delete this section:

__hassio ha info__
```json
arch: armv7
boot: true
custom: false
image: homeassistant/raspberrypi3-homeassistant
last_version: 0.90.1
machine: raspberrypi3
port: 8123
ssl: false
version: 0.90.1
wait_boot: 600
watchdog: true
```

__hassio os info__
```json
board: rpi3
version: "2.11"
version_cli: "9"
version_cli_latest: "9"
version_latest: "2.11"
core-ssh:~# hassio os info
board: rpi3
version: "2.11"
version_cli: "9"
version_cli_latest: "9"
version_latest: "2.11"

```

## Issue

__What happened:__
telemeter: Error on device update!
18:17 custom_components/telemeter/sensor.py (ERROR)
Could not decode received JSON: Expecting value: line 1 column 1 (char 0)
18:17 custom_components/telemeter/sensor.py (ERROR)

__What should have happened:__

__Possible suggestion:__

### Attachments

To help solve this issue, attach screenshots, logs, videos, gifs etc. below:

```log
```

**Comment by thibmaek** [comment]
Hey Bart, please check the info [described here](https://github.com/thibmaek/custom-components#adding-to-home-assistant) on how to add custom components to your HA instance.

It would help if you paste out the error log, now it is just grabbing in the dark.

**Comment by thibmaek** [comment]
You can update the details in the issue above

**Comment by thibmaek** [comment]
From the error log, I'm guessing your cookie probably isn't correct.
How did you grab the cookie?

**Comment by BartEngelen** [comment]
I got it from the page Telemeter inside the mijntelenet.be site

**Comment by thibmaek** [comment]
Prob. a duplicate of https://github.com/thibmaek/custom-components/issues/11
You should copy the complete cookie, which can be obtained by inspecting Doc content in the Network tab of Google Chrome/Safari/Firefox

**Comment by BartEngelen** [comment]
I use the code after AMCV_
![image](https://user-images.githubusercontent.com/17978789/55064428-41b31b80-507a-11e9-9e16-34f01ff0bf9b.png)

**Comment by thibmaek** [comment]
Please try copying the complete cookie.

**Comment by BartEngelen** [comment]
I did but still the same error in the logs:
![image](https://user-images.githubusercontent.com/17978789/55065962-17af2880-507d-11e9-8f71-d7ba46052200.png)
![image](https://user-images.githubusercontent.com/17978789/55066007-2990cb80-507d-11e9-92df-c967528b9244.png)

**Comment by thibmaek** [comment]
Could you try restarting Home Assistant and making sure you have the latest version of the Telenet component?

**Comment by BartEngelen** [comment]
I did that but still the same error.
I am on version 2.0.0 from Telemeter

**Comment by thibmaek** [comment]
Afraid I can't help, will keep this issue open but don't have the time to further check in to this for the moment.

**Comment by kallipso** [comment]
hi the same problem here

Config: 
- platform: telemeter
  name: "Telemeter"
  cookie: 'rxVisitor=15512732918XXXXXXXXXXXXXXXXXL4U6ESRIAPG55BFI7; last_visited_page=%2Fnl%2Fklantenservice%2F%23%2Fmenu%3Dselfservice%2F%23MT1; _ga=GA1.2.1054XXXXXXXX2.1556743845; lang=nl; aam_uuid=495026669XXXXXXXXXXXXX4486079863; _fbp=fb.1.1556743846448.1316670729; telenet-be-cookie-regulation=true; check=true; AMCVS_94B358XXXXXX00101%40AdobeOrg=1; AMCV_94B358XXXXXXXX87F000101%40AdobeOrg=1687686476%7CMCIDTS%7C18021%7CMCMID%7C49538433158XXXXXXXXXXXX3699418009%7CMCAAMLH-1557588500%7C6%7CMCAAMB-1557588500%7CRKhpRzXXXXXXXXXXguXWp5olkAcUniQYPHaMWWgdJ3xzPWQmdj0y%7CMCOPTOUT-1556990900s%7CNONE%7CMCAID%7CNONE%7CvVersion%7C3.0.0; dtSa=-; mbox=PC#02e9a556XXXXXXXXX45402c2.26_12#1620228501|session#631315fcb1dXXXXXXXXXXc04829916#1556985680; session_start_time=1556983820843; k_visit=2; language=nl; utag_main=v_id:016a7529e3a10001ff7111a482cXXXXXXXXXX4106a0086e$_sn:2$_ss:0$_st:15569XXXXXX$dc_u18:3$ses_id:155XXXXXXX58%3Bexp-session$_pn:3%3Bexp-session$vapi_domain:telenet.be; _gid=GAX.2.XXXXX2XX.1XXXX826; s_pers=%20s_dfa%3Dupctelenet20%7C155XXXXX492%3B%20s_cpm%3D%255B%255B%2527Direct%2527%252C%25271556983826059%2527%255D%255D%7C1714836626059%3B%20s_visitstart%3D1%7C1556985627879%3B%20s_cpdirect%3D1%7C1556985627883%3B%20gpv_pageName%3Dklantenservice%252Fhomepage%7C155XXXXXX888%3B%20gpv_prop24%3DCustomer%2520Service%7C1556985627896%3B%20s_nr%3D1556XXXXXX00-Repeat%7C1588XXXXXXX0%3B; s_sess=%20s_cc%3Dtrue%3B%20s_ppv%3D10%3B%20s_ptc%3D0.37%255E%255E0.00%255E%255E0.00%255E%255E0.00%255E%255E0.01%255E%255E0.01%255E%255E17.49%255E%255E0.42%255E%255E18.32%3B; dtPC=3$583817045_788h-vBIGESUKJCEIGHIIOKISHHDJICLIBPPMU; rxvt=1556985640511|1556983700144; dtLatC=5; BIGipServerapaaem.prd.cust.telenet.be=!uj39xSTpXZNTXRqKguZ/LkWM0Em23nLZELuPive1JKuuSX7GyGjiLh7mi3vZap8dUeQA5kOg4xr4wObDqpRe1qzSjb2+vqqoXzFusOPC; dtCookie=3$0E32CXXXXXXXXXXXXXXXXXX9A69A4B95D|437a4730db32d34a|1|6cd9XXXX8fc5a|1'



Error:
2019-05-04 17:35:55 ERROR (SyncWorker_19) [custom_components.telemeter.sensor] Could not decode received JSON: Expecting value: line 1 column 1 (char 0)
2019-05-04 17:35:55 ERROR (MainThread) [homeassistant.components.sensor] telemeter: Error on device update!
Traceback (most recent call last):
  File "/config/custom_components/telemeter/sensor.py", line 144, in make_request
    data = response.json()
  File "/usr/local/lib/python3.7/site-packages/requests/models.py", line 889, in json
    self.content.decode(encoding), **kwargs
  File "/usr/local/lib/python3.7/site-packages/simplejson/__init__.py", line 518, in loads
    return _default_decoder.decode(s)
  File "/usr/local/lib/python3.7/site-packages/simplejson/decoder.py", line 370, in decode
    obj, end = self.raw_decode(s)
  File "/usr/local/lib/python3.7/site-packages/simplejson/decoder.py", line 400, in raw_decode
    return self.scan_once(s, idx=_w(s, idx).end())
simplejson.errors.JSONDecodeError: Expecting value: line 1 column 1 (char 0)

**Comment by thibmaek** [comment]
You should immediatly change your password since pasting the cookie here exposes your credentials to everyone and they could act like they are you against Telenet services.

**Comment by kallipso** [comment]
i replaced xxxxxxxx in place of characters

**Comment by thibmaek** [comment]
If you know how, you could try using curl or Insomia or Postman with that cookie against the OCAPI url in Python and see what response you get.

**Comment by fanaticDavid** [comment]
Is this custom component still working for anyone? From time to time, it stops working but then I can fix it by replacing the cookie. For a couple of weeks now, even replacing the cookie doesn't work. This is the error I get after replacing the cookie and restarting HA:
```
2020-03-29 00:48:17 ERROR (MainThread) [homeassistant.components.sensor] telemeter: Error on device update!
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 312, in _async_add_entity
    await entity.async_device_update(warning=False)
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 476, in async_device_update
    await self.hass.async_add_executor_job(self.update)
  File "/usr/local/lib/python3.7/concurrent/futures/thread.py", line 57, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/config/custom_components/telemeter/sensor.py", line 166, in update
    self._state = response['usage_peak']
TypeError: 'NoneType' object is not subscriptable
```
Replacing the cookie and restarting HA also seems to have side effects:

- If I reopen 'Mijn Telenet' in Firefox (where I retrieved the cookie), I'm no longer logged in
- The curl command copied from within the Firefox dev tools stops working

It's as if using the cookie with this custom component invalidates the authenticated session it originated from.

**Comment by thibmaek** [comment]
I honestly haven't checked and suppose that Telenet might have changed something at the API on their side.

**Comment by fanaticDavid** [comment]
Since I'm not a Python coder and can't work out the issue myself, I decided to take an alternate route: I converted the curl command I copied from Firefox (in the developer tools) to a flow in Node-RED using an http request node. That node returns a parsed JSON object from which I extract just the values I need. Those values are then written to my MQTT broker, from where Home Assistant can pick them up with some MQTT sensors (and a template sensor). To illustrate:

<img width="767" alt="afbeelding" src="https://user-images.githubusercontent.com/20402931/78424195-ff98be80-766b-11ea-8af3-24ba4c20a147.png">

An inject node triggers this flow once an hour.
